src/_pytest/config/__init__.py

```diff
@@ -848,10 +847,13 @@ def parse(self, args, addopts=True):
                 args, self.option, namespace=self.option
             )
             if not args:
-                if self.cwd == self.rootdir:
-                    args = self.getini("testpaths")
+                if self.invocation_dir == self.rootdir:
+                    args = [
+                        str(self.invocation_dir.join(x)) if not os.path.isabs(x) else x
```

Comment:
you can use `self.invocation_dir.join(x, abs=True)` and i believe the cast to str is not needed

Reply:
The cast to string was needed for xdist.

Reply:
good point, then it is definitively needed for the line below where it sets `args = [self.invocation_dir]`